docs/examples/go-ipfs-as-a-library/main.go

```diff
@@ -0,0 +1,212 @@
+	}
+
+	// configure the temporary node
+	// cfg.Routing.Type = "dhtclient"
+	// cfg.Experimental.QUIC = true
+	// cfg.Datastore.Spec = map[string]interface{}{
+	// 	"type": "badgerds",
+	// 	"path": "badger",
+	// }
+
+	err = fsrepo.Init(repoPath, cfg)
+	if err != nil {
+		return "", fmt.Errorf("failed to init ephemeral node: %s", err)
+	}
+
+	return repoPath, nil
+}
+
+/// ------ Spawning the node
+func createNode(ctx context.Context, repoPath string) (iCore.CoreAPI, error) {
+	// Open the repo
+	repo, err := fsrepo.Open(repoPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct the node
+	node, err := core.NewNode(ctx, &core.BuildCfg{
+		Online: true,
+		// Routing: libp2p.DHTClientOption,
+		Repo: repo,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return coreapi.NewCoreAPI(node)
+}
+
+/*
+func spawnDefault(ctx context.Context) (iface.CoreAPI, error) {
+
+}
+*/
+
+/*
+func spawnEphemeral(ctx context.Context) (iface.CoreAPI, error) {
+	defaultPath, err := config.PathRoot()
+	if err != nil {
+		// shouldn't be possible
+		return nil, err
+	}
+
+	return tmpNode(ctx)
+}
+*/
+
+// Spawns a node on the default repo location, if the repo exists
+func spawnDefault(ctx context.Context) (iCore.CoreAPI, error) {
+	defaultPath, err := config.PathRoot()
+	if err != nil {
+		// shouldn't be possible
+		return nil, err
+	}
+
+	if err := setupPlugins(defaultPath); err != nil {
+		panic(err)
+	}
+
+	return createNode(ctx, defaultPath)
+}
+
+// Spawns a node to be used just for this run (i.e. creates a tmp repo)
+func spawnEphemeral(ctx context.Context) (iCore.CoreAPI, error) {
+	// Create a Temporary Repo
+	repoPath, err := createTempRepo(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temp repo: ", err)
+	}
+
+	if err := setupPlugins(repoPath); err != nil {
+		panic(err)
+	}
+
+	// Spawning an ephemeral IPFS node
+	return createNode(ctx, repoPath)
+}
+
+// ----- Writing to disk
+func writeTo(nd files.Node, fpath string) error {
+	switch nd := nd.(type) {
+	case *files.Symlink:
+		return os.Symlink(nd.Target, fpath)
+	case files.File:
+		f, err := os.Create(fpath)
+		defer f.Close()
+		if err != nil {
+			return err
+		}
+
+		var r io.Reader = nd
+		_, err = io.Copy(f, r)
+		if err != nil {
+			return err
+		}
+		return nil
+	case files.Directory:
+		err := os.Mkdir(fpath, 0777)
+		if err != nil {
+			return err
+		}
+
+		entries := nd.Entries()
+		for entries.Next() {
+			child := filepath.Join(fpath, entries.Name())
+			if err := writeTo(entries.Node(), child); err != nil {
+				return err
+			}
+		}
+		return entries.Err()
+	default:
+		return fmt.Errorf("file type %T at %q is not supported", nd, fpath)
+	}
+}
+
+/// -------
+
+func main() {
+	fmt.Println("Getting an IPFS node running")
+
+	ctx, _ := context.WithCancel(context.Background())
+
+	/*
+		fmt.Println("Spawning node on default repo")
+		ipfs, err := spawnDefault(ctx)
+		if err != nil {
+			fmt.Println("No IPFS repo available on the default path")
+		}
+	*/
```

Comment:
Following up on https://github.com/ipfs/go-ipfs/pull/6695#issuecomment-539880541. Running with default repo only works if the CID is already stored in the repo. If the CID is not available, it never returns.

Need to check if it is connecting to the network if not, why.

Reply:
It _should_ work. It may take some time to find the provider.

Reply:
I've run it for multiple minutes (10+) and it doesn't seem to make the trick. How can I go from _should work_ to understand _if it is working or not_?

Reply:
The miss was really that a node with 2048bits key can't connect to any bootstrapper.

I'm now using the IPFS Cluster pinners as the bootstrappers.